docs/guides/src/main/server/containers.adoc

```diff
@@ -46,55 +48,78 @@ ENTRYPOINT ["/opt/keycloak/bin/kc.sh"]
 ----
 The build process includes multiple stages:
 
-* The `build` command applies options and includes custom providers to create an optimized image.
+* Run the `build` command to set server build options to create an optimized image.
 * The files generated by the `build` stage are copied into a new image.
-* In this runner image, the specific run configuration is applied. That configuration contains a keystore, the environment-specific hostname configuration, and database configuration.
+* In the final image, additional configuration options for the hostname and database are set so that you don't need to set them again when running the container.
 * In the entrypoint, the `kc.sh` enables access to all the distribution sub-commands.
 
-This example uses a multi-staged build to demonstrate the build and run steps.  However, you can also build a single-staged docker image by removing the following two lines:
+When creating the container image, you are able to set any server configuration option using environment variables. For example, if you want to enable a feature like `token-exchange`, you can set the
+following environment variable to your `Dockerfile`:
```

Comment:
it is a bit unclear now imo for which image the following examples(lines 56 to 73) apply. You mention configuration options the sentence earlier(line 56), these are targeted at the "runtime image". The following examples, `token-exchange` and the custom provider example, are targeted at the `builder` image. Can you make this a bit clearer pls? If not, I guess we'd have many people running a pre-optimized image, but applying the Features / providers to the runtime image. As a result, reaug would still be run on startup, which is not what we want to advertize, right?

This was the reason I added one complete example instead of showing it fragmented btw. Make clear where what option belongs.

Reply:
I will just remove enabling features from the example. It is not really needed here because people should be aware of how build options work as per the configuration guide.

Reply:
works for me :)